xdsl/traits.py

```diff
@@ -38,26 +65,30 @@ class Pure(OpTrait):
 
 
 @dataclass(frozen=True)
-class HasParent(OpTrait):
+class HasParent(LazyOpTrait):
     """Constraint the operation to have a specific parent operation."""
 
-    parameters: tuple[type[Operation], ...]
+    _parameters: tuple[type[Operation], ...]
```

Comment:
I feel that this is a very ugly hack that makes pyright happy, but is not actually factually correct, as the `_parameters` attribute can actually be observed as being of type `Callable` at certain times in the lifetime of the object.

This just feels like lying to pyright to me, that may actually make people more confused during debugging.

Reply:
Yeah now that I think about it I think we should change it before merging. The first thing that comes to mind is that `OpTrait` doesn't actually need a `parameters` field. I think the code should work just fine if that code is deleted, @kingiler could you please try that out?

Reply:
I don't think we can just drop `parameters` field?